### PR TITLE
Add testing for server endpoint coverage in ref docs

### PR DESCRIPTION
**Pull request by adamnsch**
Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

File: doc/sphinx/source/algorithms.rst [hunk 1]
```diff
@@ -766,6 +759,52 @@ These all assume that an object of :class:`.GraphDataScience` is available as `g
...
+    Returns an estimation of the memory consumption for that procedure.
+
+.. py:function:: gds.graph.sample.rwr(graph_name: str, from_G: Graph, **config: Any)
+    -> Tuple[Graph, "Series[Any]"]
+
+    Constructs a random subgraph based on random walks with restarts.
+
+.. py:function:: gds.kcore.mutate(G: Graph, **config: Any) -> "Series[Any]"
+
+    Computes the k-core values in a network
```
Comment (adamnsch) [inline doc/sphinx/source/algorithms.rst:778]: This description is taken directly from the procedure description

File: doc/sphinx/source/graphdatascience.rst [hunk 2]
```diff
@@ -3,4 +3,5 @@ GraphDataScience
 
 .. autoclass:: graphdatascience.GraphDataScience
     :members:
-    :inherited-members:
\ No newline at end of file
+    :inherited-members:
+    :exclude-members: __init__
```
Comment (adamnsch) [inline doc/sphinx/source/graphdatascience.rst:7]: The class itself already gets the __init__ params documentation, so if we don't exclude __init__ like this they are duplicated

File: doc/sphinx/source/algorithms.rst [hunk 3]
```diff
@@ -92,14 +92,8 @@ These all assume that an object of :class:`.GraphDataScience` is available as `g
...
-.. py:function:: gds.graph.sample.cnarw(graph_name: str, from_G: Graph, **config: Any)
-    -> Tuple[Graph, "Series[Any]"]
-
-    Constructs a random subgraph based on common-neighbour-aware random walks.
-
-.. py:function:: gds.graph.sample.cnarw.estimate(G: Graph, **config: Any) -> "Series[Any]"
-
-    Returns an estimation of the memory consumption for that procedure.
+.. deprecated:: 2.4.0
+   Since GDS server version 2.4.0 you should use the endpoint :func:`gds.graph.sample.rwr` instead.
```
Comment (adamnsch) [inline doc/sphinx/source/algorithms.rst:96]: I believe this is correct use of this sphinx directive. But it does render kind of funny in our custom theme @nvitucci @recrwplay

**Comment by adamnsch** [comment]
The PR looks rather big, but that's just due to a harmless search and replace in the last commit

**Review by breakanalysis — APPROVED** [review]
LGTM :)

File: doc/sphinx/source/algorithms.rst (hunk 3 above)
Comment (recrwplay) [reply doc/sphinx/source/algorithms.rst:96]: Yes. the theme inherits the behaviour from the main docs. When a div has `class="deprecated`, we use the `::after` pseudo-class on a `<span>` inside the `<div>` to show a deprecated label. We can modify the css in the Sphinx theme to override this.

Comment (adamnsch) [reply doc/sphinx/source/algorithms.rst:96]: Ah ok :) That sounds good :+1: Is that something you want to do? Or do you need help from us?

**Pull request merged**